lib/puppet/environments.rb

```diff
@@ -115,7 +115,7 @@ def get(name)
     def get_conf(name)
       env = get(name)
       if env
-        Puppet::Settings::EnvironmentConf.static_for(env, 0, Puppet[:static_catalogs], Puppet[:rich_data])
+        Puppet::Settings::EnvironmentConf.static_for(env, Puppet[:environment_timeout], Puppet[:static_catalogs], Puppet[:rich_data])
```

Comment:
@hlindberg @thallgren FYI, it looks like this was a bug that hardcoded the environment timeout for static environments to "never cache" instead of whatever the global default was (ideally "unlimited" in production environments). Let us know if I've mischaracterized that.